### PR TITLE
installation guide for noobies :D

### PR DESCRIPTION
**Pull request by KyonLeague**
hi there, maybe you need this, to get rid off those questions: "How can I use it ?" etc. :+1:

### CODE REVIEW

**Comment by OranShuster** [comment]
for osx you should add that using brew (http://brew.sh/) is highly recommended
after installing brew just go with (brew install python) in terminal and then the rest of the guide

**Comment by Mila432** [comment]
@OranShuster Python is already installed on mac os

**Comment by OranShuster** [comment]
I know. Its still recommend installing a separate python interpreter because some package installs fail on the pre installed one. It's a known thing

**Comment by antoxxxa-2** [comment]
And what about adding logic.py and pokemon_pb2.py in a git?

File: README.md [hunk 1]
```diff
@@ -1,6 +1,16 @@
...
 Pokémon GO API in Python
 
+# Installation Guide
+
+-  Download [Python27](https://www.python.org/downloads/) _(please choose your specific system)_
+  - Under __Windows__ make sure, that Python has been added to your PATH, if not please use google to add it manually.
+-  afterwards download [pip 8.1.2](https://bootstrap.pypa.io/get-pip.py) _(use save under)_
+-  run in terminal/cmd: ``python /path/to/get-pip.py``
+-  Download the [source code](https://github.com/Mila432/Pokemon_Go_API/archive/master.zip)
+-  then install all the requirements: ``pip /path/to/requirements.txt``
```
Comment (slin1202) [inline README.md]: Shouldn't this be pip install /path/to/requirements.txt?

Comment (delold) [inline README.md]: This doesn't include the pgoapi submodule required by the API.

The repository should be cloned with `git clone --recursive https://github.com/Mila432/Pokemon_Go_API.git`

File: README.md [hunk 2]
```diff
@@ -1,6 +1,16 @@
 # Pokemon_Go_API
 Pokémon GO API in Python
 
+# Installation Guide
+
+-  Download [Python27](https://www.python.org/downloads/) _(please choose your specific system)_
+  - Under __Windows__ make sure, that Python has been added to your PATH, if not please use google to add it manually.
+-  afterwards download [pip 8.1.2](https://bootstrap.pypa.io/get-pip.py) _(use save under)_
+-  run in terminal/cmd: ``python /path/to/get-pip.py``
+-  Download the [source code](https://github.com/Mila432/Pokemon_Go_API/archive/master.zip)
```
Comment (delold) [inline README.md]: This doesn't include the pgoapi submodule required by the API (pokemon_pg2.py).

The repository should be cloned with `git clone --recursive https://github.com/Mila432/Pokemon_Go_API.git`

File: README.md (hunk 1 above)
Comment (delold) [inline README.md]: It should be `pip install -r /path/to/requirements.txt`

**Comment by Imperator26** [comment]
When I try to run main.py I get:
```
usage: main.py [-h] [-u USERNAME] [-p PASSWORD] -t TYPE -l LOCATION
main.py: error: argument -t/--type is required
```
I tried to change a few things here and there but I can't figure out how to make it run.

**Comment by fadhiilrachman** [comment]
you just type the login type, Google or PTC
type
main.py -u USERNAME -p PASSWORD -t Google/PTC -l LOCATION (ex: INDONESIA)

**Comment by Imperator26** [comment]
Right! Thanks. How is the location formatted?

**Comment by kronicd** [comment]
It seems to use Google's geocoding api. You can list a street address
between quotes.

**Comment by Imperator26** [comment]
Neat thanks!

**Comment by H0bby** [comment]
Where is logic.py and pokemon_pb2.py ?

**Comment by kronicd** [comment]
They have not been released.

The developer is considering releasing them but is concerned with the
implications of this.

You will need to wait.

**Comment by harsh1412** [comment]
Hey guys so i was trying to run this, I installed python, pip, requirements.
When i run main.py with aruguments it just logs me in show some info then return to cmd prompt. 
Any idea how to get this working?
And google login is not working for me -.-'

**Comment by kronicd** [comment]
That is all the program does at this point in time. Other features have not
been released.

**Comment by harsh1412** [comment]
Damn.. I just wasted 2 hours for nothing -.-'

**Comment by mikeres0** [comment]
@harsh1412 So many people have asked the same question as you before this. Skim through all issues open or closed before commenting... Besides, the code will more than likely be released soon.

**Comment by harsh1412** [comment]
@mikeres0 yeah i saw the comments but didn't understand any of it because of the excitement that pokemon go bot was released so fast :D

**Comment by warm200** [comment]
when did developer plan to release the feature?

**Comment by madiajijah11** [comment]
Collecting requests==2.10.0 (from -r C:\Python27\Pokemon_Go_API-master\requirements.txt (line 1))
  Using cached requests-2.10.0-py2.py3-none-any.whl
Collecting protobuf==2.6.1 (from -r C:\Python27\Pokemon_Go_API-master\requirements.txt (line 2))
  Using cached protobuf-2.6.1.tar.gz
    Complete output from command python setup.py egg_info:
    Download error on https://pypi.python.org/simple/google-apputils/: timed out -- Some packages may not be found!
    Couldn't find index page for 'google-apputils' (maybe misspelled?)
    Download error on https://pypi.python.org/simple/: timed out -- Some packages may not be found!
    No local packages or download links found for google-apputils
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "c:\users\mutian~1\appdata\local\temp\pip-build-lfkadx\protobuf\setup.py", line 200, in <module>
        "Protocol Buffers are Google's data interchange format.",
      File "c:\python27\lib\distutils\core.py", line 111, in setup
        _setup_distribution = dist = klass(attrs)
      File "c:\python27\lib\site-packages\setuptools-24.0.3-py3.5.egg\setuptools\dist.py", line 279, in __init__
      File "c:\python27\lib\site-packages\setuptools-24.0.3-py3.5.egg\setuptools\dist.py", line 323, in fetch_build_eggs
      File "c:\python27\lib\site-packages\setuptools-24.0.3-py3.5.egg\pkg_resources\__init__.py", line 826, in resolve
      File "c:\python27\lib\site-packages\setuptools-24.0.3-py3.5.egg\pkg_resources\__init__.py", line 1092, in best_match
      File "c:\python27\lib\site-packages\setuptools-24.0.3-py3.5.egg\pkg_resources\__init__.py", line 1104, in obtain
      File "c:\python27\lib\site-packages\setuptools-24.0.3-py3.5.egg\setuptools\dist.py", line 390, in fetch_build_egg
      File "c:\python27\lib\site-packages\setuptools-24.0.3-py3.5.egg\setuptools\command\easy_install.py", line 658, in easy_install
    
    distutils.errors.DistutilsError: Could not find suitable distribution for Requirement.parse('google-apputils')
    
    ----------------------------------------

got this error, how can i fix this? , any solution?
im use windows 7

**Comment by Velorianvx77** [comment]
I am having an issue with main.py, when I open it and get to where it asks for Username, I cannot type in anything.  I can press enter and it will then ask for Password, but I still can't type anything in?

**Comment by Imperator26** [comment]
You can't see it but your're writing

**Comment by harsh1412** [comment]
Anyone know how to fix the google login ? i am getting access token bad error

**Comment by slin1202** [comment]
It's likely the servers are just having issues.

**Comment by harsh1412** [comment]
@slin1202 I don't think so i tried it before like 8-9 hours ago that time i was also not able to login through google but PTC was working fine.. Can you login through google?

**Comment by kevinskyba** [comment]
I'm getting this error, access token works fine for me:

Traceback (most recent call last):
  File "main.py", line 81, in <module>
    main()
  File "main.py", line 73, in main
    dirty.start_private_show(access_token,ltype,args.location)
  File "/Users/.../Pokemon_Go_API/dirty.py", line 17, in start_private_show
    prot1=logic.gen_first_data(access_token,ltype)
  File "/Users/.../Pokemon_Go_API/logic.py", line 148, in gen_first_data
    login_request = pokemon_pb2.login_request()
AttributeError: 'module' object has no attribute 'login_request'

using the latest master right now, which has been updated an hour ago. Am I missing something?

**Comment by slin1202** [comment]
Hm, I was able to 8 hours ago. Will try once I get home,  hopefully someone
else can give an answer before then.

**Comment by harsh1412** [comment]
@slin1202 Maybe i an using a different command or some files are missing have you installed anything else which was not mention in the requirements? 
Btw i am doing this for the google : python main.py -u xxx@gmail.com -p xxx -t goo -l xxx,xxx

**Comment by slin1202** [comment]
Does just typing goo really work for type? I usually type out google, sorry
can't test it currently.

**Comment by harsh1412** [comment]
@slin1202 tried google, doesn't work.. Please let me know if you find something :/

**Comment by kevinskyba** [comment]
@harsh1412 you are just mistyping something. adding @gmail.com / @googlemail.comm to username? like not just your gmail username but your whole mail?

**Comment by Mystec** [comment]
Please add your pokemon_pb2 library to the repo! I see this file inside "old_demo" folder but it has syntax errors.

C:\Repos\Pokemon_Go_API>main.py
Traceback (most recent call last):
  File "C:\Repos\Pokemon_Go_API\main.py", line 18, in <module>
    import pokemon_pb2
ImportError: No module named pokemon_pb2

**Comment by kevinskyba** [comment]
@Mystec struggling with that one, too. Hope he will upload it. All versions I can find in other projects are missing the methods he is using.

**Comment by ch0c01d** [comment]
what i've know the **pokemon_pb2** is from **v1.0/pgoapi/old-demo/**pokemon_pb2.py
but still something is missing ~

> Traceback (most recent call last):
>   File "main.py", line 81, in <module>
>     main()
>   File "main.py", line 73, in main
>     dirty.start_private_show(access_token,ltype,args.location)
>   File "/Users/ch0c01d/Downloads/pokepokebot/dirty.py", line 19, in start_private_show
>     prot1=logic.gen_first_data(access_token,ltype)
>   File "/Users/ch0c01d/Downloads/pokepokebot/logic.py", line 155, in gen_first_data
>     login_request = pokemon_pb2.login_request()
> AttributeError: 'module' object has no attribute 'login_request'

**Comment by stealthhex** [comment]
https://github.com/NecronomiconCoding/Pokemon-Go-Rocket-API

Works like a charm

**Comment by Imperator26** [comment]
I've never compiled c# how do you do it?

**Comment by ch0c01d** [comment]
using visual studio

**Comment by Imperator26** [comment]
Yeah downloading it now

File: README.md (hunk 2 above)
Comment (kennycoc) [inline README.md]: I am missing pokemon_pg2.py even after cloning with '--recursive'

**Comment by RadBitt** [comment]
Use the files in v0.9

**Comment by djpd** [comment]
C:\Users\DJPD\Desktop\pokemon fakegps\POKEMON GO MILA PYTHON\Pokemon_Go_API-master>main.py
Traceback (most recent call last):
  File "C:\Users\DJPD\Desktop\pokemon fakegps\POKEMON GO MILA PYTHON\Pokemon_Go_API-master\main.py", line 19, in <module>
    import logic
  File "C:\Users\DJPD\Desktop\pokemon fakegps\POKEMON GO MILA PYTHON\Pokemon_Go_API-master\logic.py", line 258
    m.unknown1=
              ^
SyntaxError: invalid syntax

somebody solve?

**Comment by yaohongkok** [comment]
Hey guys,

If you read the code, it is pretty obvious that putting the logic.py in the
new source code is just a temporary thing. Whether it was a troll or not,
who knows?

Anyways, even you solve it, you still need pokemon_pb2.py. It is currently
not released by someone. :)  But, if you search online, someone got access
to that particular file. If you ask around in the web (like Reddit), you
might be able to get your hands on it.

Meanwhile, I will try the C# farming tool this weekend because they don't
seem to play little tiny tricks.  ;)

**Comment by Vanquelin** [comment]
`Microsoft Windows [Version 6.1.7601]
Copyright (c) 2009 Microsoft Corporation.  All rights reserved.

C:\Users\Xenon>cd desktop/pokemon

C:\Users\Xenon\Desktop\Pokemon>main.py -t ptc -l "Leeds University" -d 450
  File "C:\Users\Xenon\Desktop\Pokemon\main.py", line 41
    print '[!] todo from here ..'
                                ^
SyntaxError: Missing parentheses in call to 'print'

C:\Users\Xenon\Desktop\Pokemon>`
how to solve this?

**Comment by Bigtalljosh** [comment]
@Vanquelin It literally tells you right there what the problem is...

> print '[!] todo from here ..'
> ^
> SyntaxError: Missing parentheses in call to 'print'


You're missing parentheses around what is being printed...

`print( '[!] todo from here ..')`

Try that.

**Comment by crack00r** [comment]
Google login Problem 
Bad Token

**Comment by Bigtalljosh** [comment]
You are all baring in mind they just launched in Japan and the servers are being hit hard, right?

**Comment by Mila432** [comment]
@Vanquelin by using python 2.7

**Comment by Vanquelin** [comment]
Microsoft Windows [Version 6.1.7601]
Copyright (c) 2009 Microsoft Corporation.  All rights reserved.

C:\Users\Xenon>cd desktop/pokemon

C:\Users\Xenon\Desktop\Pokemon>python main.py -t ptc -l "Leeds University" -d 30
0
Traceback (most recent call last):
  File "main.py", line 5, in <module>
    from pgoapi import pgoapi
ImportError: cannot import name 'pgoapi'

C:\Users\Xenon\Desktop\Pokemon>

give me solution please

**Comment by bestkings** [comment]
at main folder, I can't login google account, it say access_token bad.
at v1.0 folder, I'm used tejado's pgoapi folder, can login now, it say todo from here. But then what I should do now?

**Comment by crack00r** [comment]
Pokestops now work, but no pokemons catching

**Comment by bestkings** [comment]
@crack00r are you login with PTC type?

**Comment by crack00r** [comment]
Nö Google

**Comment by bestkings** [comment]
I got "problem in google login, access_token bad". I don't know. I was used command: -u xxx@gmail.com -p xxx -t goo -l "New York". What wrong with me?
Someone please help me

**Comment by crack00r** [comment]
For me it was wrong password :D

**Comment by Vanquelin** [comment]
@crack00r 
Username:
Password:
[!] I am a poketrainer..
[!] doing login for: Vanquelin
[!] Your given location: Taman Mini Indonesia Indah, Ceger, Cipayung, Kota Jakar
ta Timur, Daerah Khusus Ibukota Jakarta, Indonesia
[!] lat/long/alt: -6.3024459 106.8951559 0.0
[+] Token: TGT-2105141-RElPS3Lc53HboDmwczMWhA36cIuJ...
[+] Received API endpoint: https://pgorelease.nianticlabs.com/plfe/450/rpc
[+] Login successful
[+] Username: Vanquelin
[+] You are playing Pokemon Go since: 2016-07-16 14:05:53
[+] Poke Storage: 250
[+] Item Storage: 350
[+] POKECOIN: 0
[+] STARDUST: 1100

C:\Users\Xenon\Desktop\Pokemon>
What should i do after this ?

**Comment by crack00r** [comment]
Seams you need latest update

**Comment by psam25** [comment]
I got it working but got only the output below:

`[+] found: 118 Pokestops near
[+] starting show
[!] teleport..
[!] used
[!] teleport..
[!] charging
[!] teleport..
[!] used
[!] teleport..
[!] teleport..
[!] teleport..
[!] used
[!] teleport..
[!] used
[!] charging
[!] teleport..
[!] used
[!] teleport..
[!] used
[!] teleport..
[!] teleport..
[!] used
[!] teleport..
[!] teleport..
[!] used
[?]: 0`

**Comment by Xustyx** [comment]
It would be nice to put a couple of examples using the API like pogo https://github.com/tejado/pgoapi/tree/master/examples

Thanks.

**Comment by imval** [comment]
@crack00r : did you managed to catch some pokemons ? I can farm the pookestops but not catch pokemons right now

**Comment by crack00r** [comment]
No pokemons just pokestops

**Comment by Xjkstar** [comment]
How can I do :
1、Catch Pokemon automatically
2、Drop items when bag is full

**Comment by Neex0r** [comment]
How i can start pokemon catcher?
for now i can start only pokestop farmer via main.py

**Comment by Becold** [comment]
@Mila432 Why did you remove the logic for auto-catching pokemon ? On that commit : https://github.com/Mila432/Pokemon_Go_API/commit/757c47cab6d7a4a9ed94f058056418460df839a3

**Comment by MaciekosPL** [comment]
Someone recorded by a guide on how to install it ? Please

**Comment by BenjaminSalah** [comment]
i see it teleports at the moment, is it possible to make it walk instead?

**Comment by Ceranimo** [comment]
Waiting pokemon catching and walking tips :)

**Comment by Mila432** [comment]
@BenjaminSalah it is "walking" it just prints "teleporting"

**Comment by floor114** [comment]
Can I get a ban because of this actions?

**Comment by Kolin** [comment]
@floor114 Yes. Usually temporary, though. They won't ban you from logging in but you'll find that you can't catch any Pokemon or use any pokestops. For me, it lasted for about an hour.

**Comment by floor114** [comment]
@Kolin Only for an hour? Then all functions were returned?

**Comment by Kolin** [comment]
@floor114 Yes, I'd take a temporary ban as a warning, though. I'd assume they'll start with permanent bans if you keep at it.

**Comment by BenjaminSalah** [comment]
@Mila432 so why does my egg not show an increase in distance walked? I know there is usually a delay in the updating of that information but it's been over an hour now and it still says 1.79km.

**Comment by be1ski** [comment]
~/dev/python/Pokemon_Go_API/v1.0 (master)$ python main.py -t PTC -l Moscow -u *** -p *** -l ***                                                                                               [ruby-2.0.0p648]
Traceback (most recent call last):
  File "main.py", line 5, in <module>
    from pgoapi import PGoApi
  File "/usr/local/lib/python2.7/site-packages/pgoapi/__init__.py", line 19, in <module>
    from pgoapi.pgoapi import PGoApi
  File "/usr/local/lib/python2.7/site-packages/pgoapi/pgoapi.py", line 33, in <module>
    from pgoapi.rpc_api import RpcApi
  File "/usr/local/lib/python2.7/site-packages/pgoapi/rpc_api.py", line 39, in <module>
    from . import protos
ImportError: cannot import name protos

What am I doing wrong?

**Comment by Imperator26** [comment]
What's this `use_powerball=False` in config.py?

**Comment by rizal76** [comment]
offline response?

**Comment by MaciekosPL** [comment]
How to change pokestop speed exp?

**Comment by Bokbunja** [comment]
Can anyone help me with this please? I've spent 6 hours to install everything and now I am stuck with this problem on my Command Prompt:

[ ! ] Using google as login . .
[ - ] problem in google login . .
[ - ] access_token bad

**Comment by drave** [comment]
@Bokbunja Did you create an application-specific password for your google account? [https://support.google.com/accounts/answer/185833?hl=en]

**Comment by Jefreesujit** [comment]
I m only able to collect from pokestops, but not able to spot or catch pokemons.. And even the distance walked is not getting into account.. could someone help me in this?

**Comment by quimmontal** [comment]
@drave I did that but still having the login problem, it breaks on login.py line 120, as r2 variable returns empty so there's nothing to get from there. Does anyone know where the problem resides?

**Pull request closed without merging**